components/raftstore-v2/src/operation/query/lease.rs

```diff
@@ -0,0 +1,352 @@
+        }
+
+        if !self.propose_read_index(poll_ctx, req, self.is_leader(), ch, now) {
+            return false;
+        }
+
+        // TimeoutNow has been sent out, so we need to propose explicitly to
+        // update leader lease.
+        // TODO:add following when propose is done
+        // if self.leader_lease.is_suspect() {
+        // let req = RaftCmdRequest::default();
+        // if let Ok(Either::Left(index)) = self.propose_normal(poll_ctx, req) {
+        // let (callback, _) = CmdResChannel::pair();
+        // let p = Proposal {
+        // is_conf_change: false,
+        // index,
+        // term: self.term(),
+        // cb: callback,
+        // propose_time: Some(now),
+        // must_pass_epoch_check: false,
+        // };
+        //
+        // self.post_propose(poll_ctx, p);
+        // }
+        // }
+        true
+    }
+
+    // Returns a boolean to indicate whether the `read` is proposed or not.
+    // For these cases it won't be proposed:
+    // 1. The region is in merging or splitting;
+    // 2. The message is stale and dropped by the Raft group internally;
+    // 3. There is already a read request proposed in the current lease;
+    pub fn read_index<T: Transport>(
+        &mut self,
+        poll_ctx: &mut StoreContext<EK, ER, T>,
+        mut req: RaftCmdRequest,
+        ch: QueryResChannel,
+    ) -> bool {
+        // TODO: add pre_read_index to handle splitting or merging
+        let now = monotonic_raw_now();
+        if self.is_leader() {
+            self.leader_read_index(poll_ctx, req, ch, now)
+        } else {
+            self.follower_read_index(poll_ctx, req, ch, now)
+        }
+    }
+
+    pub(crate) fn send_read_command<T>(
+        &self,
+        ctx: &mut StoreContext<EK, ER, T>,
+        read_cmd: RaftRequest<QueryResChannel>,
+    ) {
+        let mut err = errorpb::Error::default();
+        let region_id = read_cmd.request.get_header().get_region_id();
+        let read_ch = match ctx.router.send(region_id, PeerMsg::RaftQuery(read_cmd)) {
+            Ok(()) => return,
+            Err(TrySendError::Full(PeerMsg::RaftQuery(cmd))) => {
+                err.set_message(RAFTSTORE_IS_BUSY.to_owned());
+                err.mut_server_is_busy()
+                    .set_reason(RAFTSTORE_IS_BUSY.to_owned());
+                cmd.ch
+            }
+            Err(TrySendError::Disconnected(PeerMsg::RaftQuery(cmd))) => {
+                err.set_message(format!("region {} is missing", self.region_id()));
+                err.mut_region_not_found().set_region_id(self.region_id());
+                cmd.ch
+            }
+            _ => unreachable!(),
+        };
+        let mut resp = RaftCmdResponse::default();
+        resp.mut_header().set_error(err);
+        read_ch.report_error(resp);
+    }
+
+    /// response the read index request
+    ///
+    /// awake the read tasks waiting in frontend (such as unified thread pool)
+    pub(crate) fn response_read<T>(
```

Comment:
Should be `respond_read`.

Reply:
> Should be `respond_read`.

This is to be consistent with v1's code:)

Reply:
We don't need to, let's make code more readable.